### PR TITLE
fix add radio/checkbox input  multiple (with options config fields)

### PR DESCRIPTION
**Pull request by cholletNovius**
debug viewtemplate when we want add field radio or checkbox mulitple options

### CODE REVIEW

**Review by Meroje — CHANGES_REQUESTED** [review]
You replaced tabs with spaces and now the diff is all messed up and hard to follow

File: classes/fieldset/field.php [hunk 1]
```diff
@@ -583,7 +583,14 @@ protected function template($build_field)
...
 		if (is_array($build_field))
 		{
-			$label = $this->label ? str_replace('{label}', $this->label, $form->get_config('group_label', '<span>{label}</span>')) : '';
+            $tagLabel = '{label}';
+            if(is_array($this->label)) {
+                $tagLabel = array('{label}');
+            }
+
+            $labelName = $this->label ? str_replace($tagLabel, $this->label, $form->get_config('group_label', '<span>{label}</span>')) : '';
+            $label = $form->label($labelName, null, array('id' => 'label_'.$this->name));
```
Comment (cholletNovius) [inline classes/fieldset/field.php]: exemple of field config : 

``` 'prof_type' => array( 
         'label' => __('Je suis').'<span class="required">*</span>',
            'form' => array(
                'type' => 'radio',
                'value' => 'journaliste',
                'options' => array(
                    'journaliste' => __('Journaliste'),
                    'pro' => __('Professionnel'),
                ),
            ),
            'validation' => array('required'),
        ),```

when we are `validation`, `$this->label` is an array.

File: classes/fieldset/field.php [hunk 2]
```diff
@@ -602,8 +609,13 @@ protected function template($build_field)
 				return $template;
 			}
 
-			// still here? wasn't a multi field template available, try the normal one with imploded $build_field
-			$build_field = implode(' ', $build_field);
+            // map the key value $build_field (to have <input...>label)
+            $labels = array_keys($build_field);
+            $buildFieldMap = array_map(function($value, $key) {
```
Comment (cholletNovius) [inline classes/fieldset/field.php]: When radio/checkbox field is multiple like :
```
'form' => array(
                'type' => 'radio',
                'value' => 'journaliste',
                'options' => array(
                    'journaliste' => __('Journaliste'),
                    'pro' => __('Professionnel'),
                ),
            ),
```
'$build_field' is array with `'label display' => 'value input display'`
But there was no label for each input display.

**Review by Meroje — APPROVED** [review]

**Comment by cholletNovius** [comment]
up

**Comment by Meroje** [comment]
:+1: works for me

**Review by jguyomard — APPROVED** [review]
